### PR TITLE
Migrate examples from Docker to Kubernetes

### LINKED ISSUE #10: Automatically create Kafka topics in docker stack
Use init-kafka approach to automatically create topics in each Kafka related zilla example, simplifying the README accordingly as well.

### LINKED ISSUE #15: Migrate examples to use Kubernetes instead of Docker
Start by migrating the following examples to illustrate how to use Zilla with Kubernetes.

- `tcp.echo` requires configuration only, via `configmap`
- `http.echo` requires tls key and certificate, via `secret`
- `http.kafka.oneway` requires multiple containers

Then with these complete, apply a consistent approach to the remaining examples too.

### PR DESCRIPTION
**Pull request by attilakreiner**
Migrate examples from Docker to Kubernetes
fixes #15
fixes #10

### CODE REVIEW

File: http.echo/README.md [hunk 1]
```diff
@@ -3,16 +3,39 @@ Listens on http port `8080` and will echo back whatever is sent to the server.
...
+- netcat
+
+### Install zilla engine
 
-### Start zilla engine
 ```bash
-$ docker stack deploy -c stack.yml example
-Creating network example_net0
-Creating service example_zilla
+$ helm install zilla-http-echo . --namespace zilla-http-echo --create-namespace
```
Comment (jfallows) [inline http.echo/README.md:14]: Is the rationale for using a distinct namespace per example so that more than one example can be deployed at once?

Then only the example-specific port forwarding would be needed to make it work from localhost?

Comment (attilakreiner) [reply http.echo/README.md:14]: Yep, exactly.

File: http.kafka.oneway/templates/all.yaml [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+      volumes:
+      - name: config-volume
+        configMap:
+          name: zilla
+      - name: tls-volume
+        secret:
+          secretName: tls
+          items:
+          - key: localhost.p12
+            path: localhost.p12
```
Comment (jfallows) [inline http.kafka.oneway/templates/all.yaml:39]: Let's figure out our naming convention for k8s resources.

Initially, my instinct was to _not_ have the type-based suffix, such as `tls-volume` for the `volume` name. But then I looked up the k8s documentation, and found [that approach](https://kubernetes.io/docs/concepts/storage/volumes/#aws-ebs-configuration-example) is used pervasively.

So now I'm wondering, given that these are `zilla-examples`, should we be treating the k8s part as if we are teaching k8s too, with naming suffixes like `...-container`, `...-volume`, etc, or just omit the suffixes everywhere.

What do you think? IMHO we should choose a consistent strategy for all named items regardless.

**Review by jfallows — COMMENTED** [review]
Port forwarding looks great, including the killall instructions to clean up!

Please see open question about naming.

File: http.kafka.oneway/templates/all.yaml (hunk 2 above)
Comment (attilakreiner) [reply http.kafka.oneway/templates/all.yaml:39]: Here's my 2 cents. Generally speaking, I'd avoid type-based suffixes whenever they surface to the user-facing interface. The primary way users access a kubernetes cluster is through `kubectl`. For example, getting some logs:
```
$ kubectl logs --namespace zilla-http-kafka-oneway service/kafka --container zilla
# vs
$ kubectl logs --namespace zilla-http-kafka-oneway-namespace service/kafka-service --container zilla-container

```
I think the first command is really concise where the second one is quite bloated and honestly looks a bit weird.

I regard naming volumes as `something-volume` as a somewhat different case. `Deployment`s in prod can be big, complex yaml structures potentially hard to navigate/comprehend. Specifically, `volume`s (declaration) and `volumeMount`s (usage) are in different places in the yaml tree. Also, if there are multiple containers in the pod that use (some of) the same volumes, the declaration/usage will have a 1:n relationship, so volume info is really spread out to multiple places. IMO having verbose names when it comes to `volume`s is helpful for the developer to navigate a complex Deployment definition. However, this verbosity will not leak to the user-facing interface, because the scope of `volume`/`volumeMount` declarations is really just the `Deployment` where they are declared in.

To sum it up, my take is that omitting type-based suffixes should be the norm, potentially making an exception for cases where verbose names are helpful for the developer but don't result in a bloated/weird user interface. Anyway, given these are simple examples where complexity isn't an issue at this point, we can also decide to completely avoid type suffixes for the sake of a simple/consistent naming scheme.

Comment (jfallows) [reply http.kafka.oneway/templates/all.yaml:39]: Great discussion points @attilakreiner  I better understand your thought process now.

Per your last point, let's remove the suffixes as this is a simple example, but keep it in mind for more complex k8s configurations.

File: amqp.reflect/chart/templates/service-zilla.yaml [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+    matchLabels:
+      app.kubernetes.io/instance: zilla
+  template:
+    metadata:
+      labels:
+        app.kubernetes.io/instance: zilla
+    spec:
+      containers:
+        - name: zilla
+          command: ["/bin/sh", "-c", "/opt/zilla/zilla start -v -e"]
```
Comment (jfallows) [inline amqp.reflect/chart/templates/service-zilla.yaml:19]: Can we use `args` instead of `command` to rely on docker image `ENTRYPOINT` implicitly?
This will make these example k8s pod specs resilient to a change in zilla `ENTRYPOINT` location.

Suggest placing `image` and `imagePullPolicy` after `name` and before `args`.

**Review by jfallows — CHANGES_REQUESTED** [review]

File: amqp.reflect/chart/files/zilla.json [hunk 4]
```diff
@@ -11,7 +11,7 @@
                 {
                     "store": "tls/localhost.p12",
                     "type": "pkcs12",
-                    "password": "${{env.KEYSTORE_PASSWORD}}"
+                    "password": "${{env.TRUSTSTORE_PASSWORD}}"
```
[the remark this answers is not in the document]
Comment (attilakreiner) [reply amqp.reflect/chart/files/zilla.json]: done

File: amqp.reflect/chart/templates/service-zilla.yaml (hunk 3 above)
Comment (attilakreiner) [reply amqp.reflect/chart/templates/service-zilla.yaml]: done

**Review by jfallows — COMMENTED** [review]
Looks great! Ready for image pull policy changes.

**Review by jfallows — APPROVED** [review]
